src/client/game.cpp

```diff
@@ -1261,14 +1261,29 @@ void Game::shutdown()
 	chat_backend->addMessage(L"", L"");
 	m_chat_log_buf.clear();
 
+	FpsControl fps_control;
+	float indef_pos = 0;
+
 	if (client) {
 		client->Stop();
+
+		fps_control.reset();
+		f32 timer = 0.0f;
+
 		while (!client->isShutdown()) {
```

Comment:
Interesting, this loop will never run anyway. `client->Stop()` above sets `m_shutdown` to true ...

https://github.com/minetest/minetest/blob/07fe8d4481cd8fd14ac34afefcbf642df73e4b83/src/client/client.cpp#L314-L316

... and `client->isShutdown()` returns true as soon as `m_shutdown` is true:

https://github.com/minetest/minetest/blob/07fe8d4481cd8fd14ac34afefcbf642df73e4b83/src/client/client.cpp#L331-L334

Anyway, that's out of scope for this PR to investigate and fix.

Reply:
Accordingly, I've removed the changes to this loop again.

Reply:
In that case, assuming that boolean is not volatile, shouldn't the entire loop be removed?

Reply:
I'd rather not remove it without investigating first:

- what's the purpose of the code?
- did it ever work?
- why doesn't it work (anymore)?
- is there a bug because the code doesn't work?

which is likely out of scope for this PR